alws/build_planner.py

```diff
@@ -282,7 +285,10 @@ async def _add_single_ref(
                         build_index,
                         platform_dist
                     )
-                    mock_options['definitions']['dist'] = dist_macro
+                    if not dist_taken_by_user:
+                        mock_options['definitions']['dist'] = dist_macro
+                if not dist_taken_by_user and parsed_dist_macro:
+                    mock_options['definitions']['dist'] = parsed_dist_macro
```

Comment:
Should we catch the case when dist_macro is None?

Reply:
Do you mean None from the user side (UI, API etc) ? Or automatically parsing for modules? (In case when it is not module build we catch the case)

Reply:
Do you mean None from the user side (UI, API etc) ? Or automatically parsing for modules? (In case when it is not module build we catch the case)

Reply:
I mean the 2nd case

Reply:
As I see the code of calc_dist_macro() it cannot return None